packages/storage/src/indexed-db/index.ts

```diff
@@ -212,6 +214,22 @@ export class IndexedDb implements IndexedDbInterface {
     });
   }
 
+  async getAppParams(): Promise<AppParameters | undefined> {
+    const json = await this.db.get('APP_PARAMETERS', 'params');
+    if (!json) return undefined;
+    return AppParameters.fromJson(json);
+  }
+
+  async saveAppParams(app: AppParameters): Promise<void> {
+    // this always fails, because the chainId is corrupted
+    //if (app.chainId != this.c.name) throw new Error(`New chainId mismatch: stored ${this.c.name}, new ${app.chainId}`);
+    await this.u.update({
+      table: 'APP_PARAMETERS',
+      value: app.toJson() as Jsonified<AppParameters>,
```

Comment:
I suppose this is where we're noticing the missing shielded pool field, right?

Reply:
we aren't using the shieldedPool field for anything yet. but anywhere we make an appParameters request to grpc, the deserialized response is corrupt. right now it matters for the chainId which we get from that endpoint.

Reply:
we do use sctParams from this endpoint, when asking wasm to plan a tx. it seems that 'epochDuration' is deserialized as `1n`, which doesn't seem correct